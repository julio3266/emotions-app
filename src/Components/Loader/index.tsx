import React from 'react';
import * as Styled from './styles';

export const Loader: React.FC = () => {
  return (
    <Styled.Container>
      <></>
    </Styled.Container>
  );
};
