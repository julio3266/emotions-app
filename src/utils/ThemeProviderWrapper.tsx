import React from 'react';
import {ThemeProvider} from 'styled-components/native';
import Theme from '../theme/styles';

interface IProps {
  children: any;
  padding?: boolean;
}

export const ThemeProviderWrapper: React.FC<IProps> = ({children}) => {
  return <ThemeProvider theme={Theme}>{children}</ThemeProvider>;
};
