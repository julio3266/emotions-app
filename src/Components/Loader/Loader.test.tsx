import React from 'react';
import {ThemeProviderWrapper as wrapper} from '@utils/ThemeProviderWrapper';
import {mockTestID} from '@utils/qualityAssurance';
import {render} from '@testing-library/react-native';
import {Loader, ILoaderProps} from '@Components/Loader';

describe('Components/Loader', () => {
  let props: ILoaderProps = {
    testID: 'Loader',
  };

  const {findByTestId, toJSON} = render(<Loader {...props} />, {
    wrapper,
  });
  const sut_container = mockTestID('View', props.testID!).testID;
  const sut_spinner = mockTestID(
    'ActivityIndicator',
    `${props.testID!}_spinner`,
  ).testID;
  it(`should render #${sut_container}`, () => {
    const sut = findByTestId(sut_container);
    expect(sut).toBeTruthy();
  });

  it(`should render #${sut_spinner}`, () => {
    const sut = findByTestId(sut_spinner);
    expect(sut).toBeTruthy();
  });

  it(`should render a snapshot`, () => {
    expect(toJSON()).toMatchSnapshot();
  });
});
