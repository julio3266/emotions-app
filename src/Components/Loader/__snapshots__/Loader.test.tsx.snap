// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Loader should render a snapshot 1`] = `null`;
